x-pack/plugins/ml/public/jobs/new_job_new/common/job_creator/single_metric_job_creator.ts

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { parseInterval } from 'ui/utils/parse_interval';
+import { JobCreator } from './job_creator';
+import { Field, Aggregation } from '../../../../../common/types/fields';
+import { Detector, BucketSpan } from './configs';
+import { createBasicDetector } from './util/default_configs';
+
+export class SingleMetricJobCreator extends JobCreator {
+  private _field: Field | null = null;
+  private _agg: Aggregation | null = null;
+
+  // only a single detector exists for this job type
+  // therefore _addDetector and _editDetector merge into this
+  // single setDetector function
+  public setDetector(agg: Aggregation, field: Field | null) {
+    const dtr: Detector = createBasicDetector(agg, field);
+
+    if (this._detectors.length === 0) {
+      this._addDetector(dtr);
+    } else {
+      this._editDetector(dtr, 0);
+    }
+
+    this._field = field;
+    this._agg = agg;
+
+    this._createAggregations();
+  }
+
+  public set bucketSpan(bucketSpan: BucketSpan) {
+    this._job_config.analysis_config.bucket_span = bucketSpan;
+    this._createAggregations();
+  }
+
+  // aggregations need to be recreated whenever the detector or bucket_span change
+  private _createAggregations() {
+    if (
+      this._detectors.length &&
+      typeof this._job_config.analysis_config.bucket_span === 'string' &&
+      this._agg !== null
+    ) {
+      delete this._job_config.analysis_config.summary_count_field_name;
+      delete this._datafeed_config.aggregations;
+
+      const functionName = this._agg.dslName;
+      const timeField = this._job_config.data_description.time_field;
+
+      const duration = parseInterval(this._job_config.analysis_config.bucket_span);
+      if (duration === null) {
+        return;
+      }
+
+      const bucketSpanSeconds = duration.asSeconds();
+      const interval = bucketSpanSeconds * 1000;
+
+      switch (functionName) {
+        case 'count':
```

Comment:
It's worth creating enums for these function names I think as they will be used elsewhere.